src/Admin/DevToolsUserAccess.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Class DevToolsUserAccess.
+ *
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_User;
+
+/**
+ * Class DevToolsUserAccess
+ *
+ * @since 1.6.0
+ */
+final class DevToolsUserAccess implements Delayed, Service, Registerable {
+
+	/**
+	 * User meta key enabling or disabling developer tools.
+	 *
+	 * @var string
+	 */
+	const USER_FIELD_DEVELOPER_TOOLS_ENABLED = 'amp_dev_tools_enabled';
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Runs on instantiation.
+	 *
+	 * @action rest_api_init
+	 */
+	public function register() {
+		register_rest_field(
+			'user',
+			self::USER_FIELD_DEVELOPER_TOOLS_ENABLED,
+			[
+				'get_callback'    => [ $this, 'rest_get_dev_tools_enabled' ],
+				'update_callback' => [ $this, 'rest_update_dev_tools_enabled' ],
+				'schema'          => [
+					'description' => __( 'Whether the user has enabled dev tools.', 'amp' ),
+					'type'        => 'boolean',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Provides the user's dev tools enabled setting.
+	 *
+	 * @param array $user Array of user data prepared for REST.
+	 * @return null|boolean Whether tools are enabled for the user, or null if the option has not been set.
+	 */
+	public function rest_get_dev_tools_enabled( $user ) {
+		$meta = get_user_meta( $user['id'] );
+
+		if ( is_array( $meta ) && array_key_exists( self::USER_FIELD_DEVELOPER_TOOLS_ENABLED, $meta ) ) {
+			return boolval(
+				is_array( $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] ) && ! empty( $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] )
+					? reset( $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] )
+					: $meta[ self::USER_FIELD_DEVELOPER_TOOLS_ENABLED ]
+			);
+		}
+
+		// If the field is not yet set, don't make a default selection in the setup wizard.
+		return null;
+	}
+
+	/**
+	 * Updates a user's dev tools enabled setting.
+	 *
+	 * @param bool    $new_value New setting for whether dev tools are enabled for the user.
+	 * @param WP_User $user      The WP user to update.
+	 * @return int|bool|WP_Error The result of update_user_meta, or WP_Error if the current user lacks permission.
+	 */
+	public function rest_update_dev_tools_enabled( $new_value, WP_User $user ) {
+		if ( ! current_user_can( 'manage_options' ) || ! current_user_can( 'edit_user', $user->ID ) ) {
+			return new WP_Error(
+				'amp_rest_cannot_edit_user',
+				__( 'Sorry, the current user is not allowed to make this change.', 'amp' ),
+				[ 'status' => rest_authorization_required_code() ]
+			);
+		}
+
+		return update_user_meta( $user->ID, self::USER_FIELD_DEVELOPER_TOOLS_ENABLED, boolval( $new_value ) );
```

Comment:
Rather than storing a boolean, WordPress core stores a string `"true"` and `"false"` for user options:

![image](https://user-images.githubusercontent.com/134745/85192141-4911a680-b276-11ea-8208-b6b4d7beb99b.png)

This gets around the problem of storing  a true boolean value in a MySQL `TEXT` field. I think we should do the same.

```suggestion
		return update_user_meta( $user->ID, self::USER_FIELD_DEVELOPER_TOOLS_ENABLED, wp_json_encode( (bool) $new_value );
```

Reply:
Done in 3139da785878c7300d8afc5a3e48b878bbbc11e4.

Something to think about is that when a user is created in WordPress, it will by default supply all the user meta for, including those boolean options. So it will set the meta for `rich_editing`, `syntax_highlighting`, `show_admin_bar_front`, and `comment_shortcuts`. See the code in `wp_insert_user()`: https://github.com/WordPress/wordpress-develop/blob/1db25e3d3841e47e07eb71bf2b2c609cb37e71a1/src/wp-includes/user.php#L1763-L1774

That being the case, it's _different_ that we are not providing an initial value here. Nevertheless, I think it's probably the right call because we don't know what the user's default choice should be.